src/kspace.cpp

```diff
@@ -443,15 +443,23 @@ void KSpace::modify_params(int narg, char **arg)
       nx_pppm = nx_msm_max = force->inumeric(FLERR,arg[iarg+1]);
       ny_pppm = ny_msm_max = force->inumeric(FLERR,arg[iarg+2]);
       nz_pppm = nz_msm_max = force->inumeric(FLERR,arg[iarg+3]);
-      if (nx_pppm == 0 && ny_pppm == 0 && nz_pppm == 0) gridflag = 0;
+      if (nx_pppm == 0 && ny_pppm == 0 && nz_pppm == 0)
```

Comment:
I don't think we need this check. The default (if no `kspace_modify mesh` is given) is to set the grid automatically, so I don't think a user would go to the trouble of explicitly setting the grid to all zeros using a `kspace_modify` command, just to get the default behavior.

Reply:
you never know what is going to happen in the future and what people are going to try and then get confused by inconsistent behavior of a program. this extra test doesn't cost us any performance and makes certain, that settings are always consistent and people can switch around as they like.

Reply:
Is this documented anywhere though?

Reply:
Yes it is documented in kspace_modify, "Values for x,y,z of 0,0,0 unset the option." So you are right, we should leave it.